### ISSUE TITLE
Amazon Linux support missing

### ISSUE THREAD

**Issue by eric-tucker** [chef-cookbooks/docker#812]
### Cookbook version
2.13.11

### Chef-client version
12.5.1

### Platform Details
Amazon Linux 2016.09

### Scenario:
Can't install Docker through docker_service as curl -sSL https://get.docker.com/ | sh does not work

### Steps to Reproduce:
Run docker_service resource against Amazon Linux

### Expected Result:
Installs  :)

### Actual Result:
```   ================================================================================
    Error executing action `create` on resource 'docker_service[default]'
    ================================================================================
    
    Mixlib::ShellOut::ShellCommandFailed
    ------------------------------------
    docker_installation_script[default] (/etc/chef/local-mode-cache/cache/cookbooks/docker/libraries/docker_service.rb line 57) had an error: Mixlib::ShellOut::ShellCommandFailed: execute[install docker] (/etc/chef/local-mode-cache/cache/cookbooks/docker/libraries/docker_installation_script.rb line 39) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
    ---- Begin output of curl -sSL https://get.docker.com/ | sh ----
    STDOUT: 
    STDERR: Either your platform is not easily detectable, is not supported by this
      installer script (yet - PRs welcome! [hack/install.sh]), or does not yet have
      a package for Docker.  Please visit the following URL for more detailed
      installation instructions:
    
        https://docs.docker.com/engine/installation/
    ---- End output of curl -sSL https://get.docker.com/ | sh ----
    Ran curl -sSL https://get.docker.com/ | sh returned 1
    
    Cookbook Trace:
    ---------------
    /etc/chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:78:in `run_action'
    /etc/chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block (2 levels) in converge'
    /etc/chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `each'
    /etc/chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block in converge'
    /etc/chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:105:in `converge'
    /etc/chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/provider.rb:133:in `compile_and_converge_action'
    /etc/chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:78:in `run_action'
    /etc/chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block (2 levels) in converge'
    /etc/chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `each'
    /etc/chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block in converge'
    /etc/chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:105:in `converge'
    /etc/chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/provider.rb:133:in `compile_and_converge_action'
    /etc/chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:78:in `run_action'
    /etc/chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block (2 levels) in converge'
    /etc/chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `each'
    /etc/chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block in converge'
    /etc/chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:105:in `converge'
    
    Resource Declaration:
    ---------------------
    # In /etc/chef/local-mode-cache/cache/cookbooks/my_cookbook/recipes/batch_server.rb
    
     56: docker_service "default" do
     57:   action [:create, :start]
     58: end
     59: 
```

This has been broken since the 1.12 release because of this:

https://github.com/docker/docker/issues/27313

I can't believe no one has complained about this yet.  :(

**Comment by eric-tucker** [comment]
Side note: docker 1.12.6 is available in the Amazon Linux repositories.  The cookbook may have to use this as the installation source until this has been resolved.

**Comment by eric-tucker** [comment]
@someara I've seen you do commits and publish 2.14.0.  I've just tested this and it still does not work.  The cookbook continues to attempt to use the script at get.docker.com which has removed Amazon Linux support.  You need to make the cookbook use the package install method.

**Comment by someara** [comment]
The test recipes reference docker_installation_package directly so we can be sure of the version.
That means I missed the auto-select testing for amazon, since it's not included in the "auto" suites.

Should be fixed up  979cdfbfbc3fcc364a3ef301c87b80b0e505a87d
